packages/datetime/src/dateInput.tsx

```diff
@@ -320,4 +321,36 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     private setInputRef = (el: HTMLElement) => {
         this.inputRef = el;
     }
+
+    /**
+     * Translate a moment into a Date object, adjusting the moment timezone into the local one.
```

Comment:
Date objects are always in the local timezone. Moment operates in the specified timezone after moment.tz.setDefault has been called (or the local timezone if none has been configured). Therefore directly copying the individual components of the dates corrects for the timezone discrepancy.